autogpt_platform/backend/backend/server/external/routes/v1.py

```diff
@@ -97,13 +97,13 @@ async def execute_graph(
     api_key: APIKey = Depends(require_permission(APIKeyPermission.EXECUTE_GRAPH)),
 ) -> dict[str, Any]:
     try:
-        graph_exec = await internal_api_routes.execute_graph(
+        graph_exec = add_graph_execution(
```

Comment:
I've done this in a couple of places @majdyz, everywhere (including the `execute_graph` endpoint itself) now uses `.executor.utils.add_graph_execution(..)`.

Reply:
We can't, you can move the code for execute_graph endpoint somewhere else as a fully async shrrable code and use it if you want. But we should never use blocking I/O call in async code.

Reply:
If you literally have no other option, then make the route sync, or use to_thread or spin a new thread for handling the request. But I'd like to avoid this as much as possible.